gradebookng/tool/src/java/org/sakaiproject/gradebookng/business/GradebookNgBusinessService.java

```diff
@@ -1334,47 +1354,32 @@ public void putAssignmentsAndCategoryItemsInMatrix(Map<String, GbStudentGradeInf
 
 						boolean gradeable = false;
 
-						for (final PermissionDefinition permission : permissions) {
-							// we know they are all GRADE so no need to check here
-
-							boolean categoryOk = false;
-							boolean groupOk = false;
+						//If TA does not have all the permissions over categories and groups
+						if (!allPermissions) {
 
-							final Long permissionCategoryId = permission.getCategoryId();
-							final String permissionGroupReference = permission.getGroupReference();
+							//Check category and group permissions
+							List<String> arr = permByCat.get(gradeCategoryId) != null ? permByCat.get(gradeCategoryId) : new ArrayList<String>();
```

Comment:
You're doing two map lookups here when you don't need to.

List\<String\> arr = permByCat.getOrDefault(gradeCategoryId, new ArrayList\<String\>());